lib/hash_sockaddr.c

```diff
@@ -96,11 +96,11 @@ void *mptcpd_hash_sockaddr_key_copy(void const *p)
         if (src->sa->sa_family == AF_INET) {
                 sa = (struct sockaddr *) l_new(struct sockaddr_in, 1);
 
-                memcpy(sa, src->sa, sizeof(struct sockaddr_in));
+                memcpy(sa, src->sa, sizeof(*src->sa));
```

Comment:
Bear in mind that `*src->sa` in this context is a `struct sockaddr`.

This would work for IPv4 addresses represented by a `struct sockaddr_in` but the way `struct sockaddr` is used in the proposed change is generally not how `struct sockaddr` is used.  A pointer to a specific socket address type, such as `struct sockaddr_in *` is meant to be cast to `struct sockaddr *` for use in the socket API functions in the C library, for example.  Mptcpd adopted this admittedly awkward interface as well.